src/commands/PublishCommand.js

```diff
@@ -536,12 +536,26 @@ export default class PublishCommand extends Command {
       this.updatePackageDepsObject(pkg, "devDependencies", exact);
       this.updatePackageDepsObject(pkg, "peerDependencies", exact);
 
+      // exec preversion script
+      pkg.runScript("preversion", (err) => {
```

Comment:
This will run script asynchronously, wouldn't it? Same goes for other hooks that were added.
Are we fine with that?

Reply:
That’s true. I forgot that almost everything in publish is synchronous, unlike bootstrap where parallelism is important.